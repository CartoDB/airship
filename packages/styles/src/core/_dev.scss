@import 'tools/_mixins';
@import 'variables/_variables';
