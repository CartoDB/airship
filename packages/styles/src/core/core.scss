@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
