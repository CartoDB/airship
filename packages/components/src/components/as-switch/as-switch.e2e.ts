import { EventSpy } from '@stencil/core/dist/declarations';
import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-switch', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage({ html: `<as-switch id="switch" label="Switch Label"></as-switch>` });
    });

    it('should render label if prop is present', async () => {
      // TODO: Test
      const element = await page.find('as-switch');
      expect(element.outerHTML).toMatchSnapshot();
    });
  });

  describe('when the component is enabled', () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage({ html: `<as-switch id="switch"></as-switch>` });
    });

    it('should emit an event with current status when component toggles false->true', async () => {
      const changeSpy = await page.spyOnEvent('change');

      await page.click('as-switch');
      await page.waitForChanges();

      expect(changeSpy).toHaveReceivedEventDetail(true);
    });

    it('should emit an event with current status when component toggles false->true->false->true', async () => {
      const changeSpy: EventSpy = await page.spyOnEvent('change');

      const element = await page.find('as-switch');
      element.click();
      element.click();
      element.click();
      await page.waitForChanges();

      expect(changeSpy).toHaveReceivedEventTimes(3);
      expect(changeSpy.events[0].detail).toBe(true);
      expect(changeSpy.events[1].detail).toBe(false);
      expect(changeSpy.events[2].detail).toBe(true);
    });
  });

  describe('when the component is disabled', () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage({ html: `<as-switch disabled id="switch"></as-switch>` });
    });

    it('should NOT emit an event with current status when component toggles', async () => {
      const changeSpy: EventSpy = await page.spyOnEvent('change');

      await page.click('as-switch');
      await page.waitForChanges();

      expect(changeSpy).not.toHaveReceivedEvent();
    });
  });

  describe('when the component is checked', () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage({ html: `<as-switch checked id="switch"></as-switch>` });
    });

    it('should return event (FALSE) when the element is clicked for the first time', async () => {
      const changeSpy: EventSpy = await page.spyOnEvent('change');

      await page.click('as-switch');
      await page.waitForChanges();

      expect(changeSpy).toHaveReceivedEventDetail(false);
    });
  });
});
