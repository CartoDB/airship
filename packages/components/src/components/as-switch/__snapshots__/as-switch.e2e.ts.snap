// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-switch Rendering should render label if prop is present 1`] = `"<as-switch id=\\"switch\\" label=\\"Switch Label\\" class=\\"hydrated\\"><div class=\\"as-switch\\"><div class=\\"as-switch__element\\"><input class=\\"as-switch__input\\" role=\\"switch\\" id=\\"switch\\" type=\\"checkbox\\"><label class=\\"as-switch__toggle\\" for=\\"switch\\"></label><svg class=\\"as-switch__thumb\\" width=\\"10\\" height=\\"8\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"currentColor\\" d=\\"M3.315 7.858L.133 4.441a.506.506 0 0 1 0-.683l.643-.684a.437.437 0 0 1 .642 0l2.219 2.393L8.58.141a.437.437 0 0 1 .643 0l.643.683a.504.504 0 0 1 0 .683l-5.91 6.35a.437.437 0 0 1-.642 0\\"></path></svg></div><label class=\\"as-switch__label as-body\\">Switch Label</label></div></as-switch>"`;
